mysql-test/t/gap_lock_error_memory_myisam.test

```diff
@@ -0,0 +1,17 @@
+# Non transactional engines should ignore gap lock errors
+
+let $engine=memory;
```

Comment:
Again, inherited from upstream patch. Filename says both myisam and memory, it tests both engines. Maybe should be split up into two different tests.